src/dft/backends/cufft/CMakeLists.txt

```diff
@@ -32,8 +32,13 @@ add_library(${LIB_OBJ} OBJECT
 add_dependencies(onemkl_backend_libs_dft ${LIB_NAME})
 
 target_include_directories(${LIB_OBJ}
-  PRIVATE ${PROJECT_SOURCE_DIR}/include
-          ${PROJECT_SOURCE_DIR}/src
+  PUBLIC ${ONEMKL_INTERFACE_INCLUDE_DIRS}
+)
+target_include_directories(${LIB_NAME}
+  PUBLIC ${ONEMKL_INTERFACE_INCLUDE_DIRS}
+)
```

Comment:
Both of these are required, but I don't understand why. Later on there is `target_link_libraries(${LIB_NAME} PUBLIC ${LIB_OBJ})` which should make the second of these redundant.

If anyone could clear up my confusion, I'd be greatful!

Reply:
I don't get why this doesn't work either. I wonder if it can be related to the fact that `${LIB_OBJ}` is an `OBJECT` target which maybe are not be meant to have an include dir property?
Some of this is discussed there: https://stackoverflow.com/questions/38832528/transitive-target-include-directories-on-object-libraries.

Reply:
That seems like a reasonable explanation. If anyone thinks there is a better way of doing this, they can open a new comment.